File: advent-of-code/2021/day-2/Rust/day-2/src/main.rs
use std::fs;

use day_2::try_get_position;

use anyhow::Result;

// start 12:10
// read file (they are relative to src) 12:23
// split magnitude and direction 12:35
// done part 1 12:54
// done part 2 12:58
// refactor 1:06
// total time 0:54
fn main() -> Result<()> {
    let instructions = fs::read_to_string("../../input.txt")?;

    let (horizontal, depth, _) = try_get_position(instructions)?;

    println!("answer: {}", horizontal * depth);

    Ok(())
}

File: advent-of-code/2021/day-2/Go/cmd/day-2/main.go
package main

import (
	"fmt"
	"os"
)

// start 7:18
// reads file relative to go.mod 7:24
func main() {
	bytes, err := os.ReadFile("../input.txt")

	if err != nil {
		fmt.Print(err)
	}

	str := string(bytes)

	fmt.Println(str)
}
